### PR TITLE
storage: Enable on-demand installation of Stratis on RHEL 9

### PR DESCRIPTION
**Pull request by mvollmer**
This is only for rhel-9-3.

- [x] https://github.com/cockpit-project/bots/pull/4639

### CODE REVIEW

**Comment by mvollmer** [comment]
As @martinpitt has pointed out, we could use `PLATFORM_ID`. I think I would just use it literally without parsing it:

```
{
   ...
   "stratis-package": {
       "platform:el9": "stratisd"
   } 
}
```

File: pkg/storaged/client.js [hunk 1]
```diff
@@ -1287,12 +1287,20 @@ client.wait_for = function wait_for(cond) {
...
 client.get_config = (name, def) => {
     if (cockpit.manifests.storage && cockpit.manifests.storage.config) {
-        let val = cockpit.manifests.storage.config[name];
+        const val = cockpit.manifests.storage.config[name];
         if (typeof val === 'object' && val !== null)
-            val = val[client.os_release.ID];
-        return val !== undefined ? val : def;
+            return try_fields(val, [client.os_release.PLATFORM_ID, client.os_release.ID], def);
+        else
+            return val !== undefined ? val : def;
```
Comment (cockpituous) [inline pkg/storaged/client.js:1303]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18529-20230322-102433-b6457f45-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by drckeefe** [comment]
@mvollmer this would install Straits when storaged package is install or when Straits pool create is issued?

**Comment by mvollmer** [comment]
> @mvollmer this would install Straits when storaged package is install or when Straits pool create is issued?

It will be installed when creating the first pool.

**Comment by mvollmer** [comment]
Cockpit main is now going to rhel-9-3, so this is no longer blocked. @marusak, that's correct, right?

**Comment by martinpitt** [comment]
@mvollmer Correct -- 9.2 is "done" from our upstream POV.

File: pkg/storaged/client.js (hunk 1 above)
Comment (cockpituous) [inline pkg/storaged/client.js:1303]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18529-20230414-075147-975bec03-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by mvollmer** [comment]
The rhel-8-9 failure should be fixed by #18638.

File: pkg/storaged/client.js [hunk 2]
```diff
@@ -1286,12 +1286,20 @@ client.wait_for = function wait_for(cond) {
...
```
Comment (cockpituous) [inline pkg/storaged/client.js:1302]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18529-20230425-104804-82e2505d-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: test/verify/check-storage-stratis [hunk 3]
```diff
@@ -438,36 +438,43 @@ class TestStorageStratis(StorageCase):
...
-                        """{ "config": { "stratis_package": "fake-stratisd" } }""")
-        self.createPackage("fake-stratisd", "999", "1")
-        self.enableRepo()
-        m.execute("pkcon refresh")
+        # RHEL 8 should not offer installation of Stratis from Cockpit
+        # itself.
+        #
+        ondemand_stratis = "rhel-8" not in m.image
+
+        m.execute("systemctl stop stratisd && dnf remove -y stratisd stratis")
```
Comment (martinpitt) [inline test/verify/check-storage-stratis:452]: Please avoid `&&`, it's semantics are really confusing. `m.execute()` runs with `set -e`, so using `;` does the right thing.

**Review by martinpitt — APPROVED** [review]
Thanks! Agreed, let's land this in 9.3 devel ASAP.

**Pull request merged**